### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by KVelasco**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -29,3 +29,6 @@ POM_LICENCE_DIST=repo
 POM_DEVELOPER_ID=twyatt
 POM_DEVELOPER_NAME=Travis Wyatt
 POM_DEVELOPER_URL=https://github.com/twyatt
+
+kotlin.js.webpack.major.version=5
+kotlin.js.compiler=both
```
Comment (KVelasco) [inline gradle.properties:34]: do we care about supporting both legacy and ir? @twyatt @cedrickcooke

Comment (twyatt) [reply gradle.properties:34]: If it isn't a big lift, then I think we'll want to support legacy for this library. If it's causes you trouble though, I'm also fine with going with IR only.

File: build.gradle.kts [hunk 2]
```diff
@@ -4,26 +4,33 @@ import java.net.URI
...
-    id("binary-compatibility-validator") version "0.2.3"
+    kotlin("multiplatform") version "1.5.20" apply false
+    id("com.android.application") version "4.2.0" apply false
+    id("com.android.library") version "4.2.0" apply false
+    id("kotlinx-atomicfu") version "0.16.2" apply false
+    id("org.jmailen.kotlinter") version "3.4.5" apply false
     id("org.jetbrains.dokka") version "1.4.30"
-    id("com.vanniktech.maven.publish") version "0.14.0" apply false
+    id("com.vanniktech.maven.publish") version "0.17.0" apply false
     id("net.mbonnin.one.eight") version "0.1"
```
Comment (twyatt) [inline build.gradle.kts:19]: Might as well bump this to 0.2.

File: gradle.properties (hunk 1 above)
Comment (KVelasco) [reply gradle.properties:34]: sounds good! I will leave it as then and support `Legacy` and `IR`

File: build.gradle.kts [hunk 3]
```diff
@@ -4,26 +4,33 @@ import java.net.URI
...
+    // Disabling until https://github.com/Kotlin/binary-compatibility-validator/issues/38 is fixed.
+    //    id("binary-compatibility-validator") version "0.6.0"
 }
 
 allprojects {
     repositories {
         google()
-        jcenter()
+        mavenCentral()
+        maven(url = "https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven")
```
Comment (twyatt) [inline build.gradle.kts:33]: nit: for consistency, drop the name for the parameter.

```suggestion
        maven("https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven")
```

**Review by twyatt — APPROVED** [review]

**Review by cedrickcooke — APPROVED** [review]

**Pull request merged**